pull-request/src/app.js

```diff
@@ -7,7 +7,12 @@ var Person = function(first, last, middle) {
 Person.prototype = {
 
     whoAreYou : function() {
-        return this.first + (this.middle ? ' ' + this.middle: '') + ' ' + this.last;
+        if (this.first === NaN) {
+          return this.last;
+          i = 0;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move the declaration of "i" before this usage. ([javascript:VariableDeclarationAfterUsage](http://localhost:9750/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage))

[See it in SonarQube](http://localhost:9750/project/issues?id=training%3Apull-request&pullRequest=25&issues=AWgPfv8P6PH_c9BW1DXm&open=AWgPfv8P6PH_c9BW1DXm)

[ISSUE_KEY]: AWgPfv8P6PH_c9BW1DXm